### ISSUE TITLE
[Core] Projects class - Add not null and unique constraint to the name column of the Project table

### ISSUE THREAD

**Issue by laemtl** [aces/Loris#6479]
**master branch**
 [/php/libraries/Project.class.inc:createNew()](https://github.com/aces/Loris/blob/master/php/libraries/Project.class.inc#L141)
Add the not null / unique constraints on the name column of the Project table and remove the below comments.

```
// TODO :: Add not null constraint to the name column of the Project table
if (empty($projectName)) {
   throw new \LorisException("ProjectName can't be null");
}

// TODO :: Add unique constraint to the name column of the Project table
$projectId = $factory->database()->pselectOne(...)
```
See #5687 for more details

**Comment by christinerogers** [comment]
Here's how I'm tweaking your issue: 
- [ ] specify which branch : in this case `master`
- [ ] add links directly to the code.

**Comment by laemtl** [comment]
Fixed by #8243 and #8295
